DocumentDataAPI/DocumentDataAPI/Data/Repositories/NpgSimilarDocumentRepository.cs

```diff
@@ -0,0 +1,92 @@
+using System.Data;
+
+namespace DocumentDataAPI.Data.Repositories;
+
+class NpgSimilarDocumentRepository : ISimilarDocumentRepository
+{
+    private readonly IDbConnectionFactory _connectionFactory;
+    private readonly ILogger<NpgSimilarDocumentRepository> _logger;
+    private readonly ISqlHelper _sqlHelper;
+    public NpgSimilarDocumentRepository(IDbConnectionFactory connectionFactory, ILogger<NpgSimilarDocumentRepository> logger, ISqlHelper sqlHelper)
+    {
+        _connectionFactory = connectionFactory;
+        _logger = logger;
+        _sqlHelper = sqlHelper;
+    }
+
+    public async Task<long> Add(SimilarDocumentModel entity)
+    {
+        _logger.LogDebug("Adding Source with mainDocumentId: {mainDocumentId} and" +
+            " similarDocumentId: {similarDocumentId} to database", entity.MainDocumentId, entity.SimilarDocumentId);
+        _logger.LogTrace("similarDocument: {similarDocument}", entity);
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        return await con.QuerySingleAsync<long>($"insert into similar_documents({SimilarDocumentMap.MainDocumentId}, {SimilarDocumentMap.SimilarDocumentId}) " +
+            $"values (@MainDocumentId, @SimilarDocumentId) returning {SimilarDocumentMap.MainDocumentId}",
+            new { entity.MainDocumentId, entity.SimilarDocumentId });
+    }
+
+    public async Task<IEnumerable<long>> AddBatch(List<SimilarDocumentModel> models)
+    {
+        IEnumerable<long> allInsertedIds = new List<long>();
+        _logger.LogDebug("Adding {count} SimilarDocument to database", models.Count);
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        con.Open();
+        using IDbTransaction transaction = con.BeginTransaction();
+        try
+        {
+            // Divide the list of models into chunks to keep the INSERT statements from getting too large.
+            foreach (SimilarDocumentModel[] chunk in models.Chunk(_sqlHelper.InsertStatementChunkSize))
+            {
+                string parameterString = _sqlHelper.GetBatchInsertParameters(chunk, out Dictionary<string, dynamic> parameters);
+                IEnumerable<long> insertedIds = await transaction.QueryAsync<long>(
+                        $"insert into similar_documents({SimilarDocumentMap.MainDocumentId}, {SimilarDocumentMap.SimilarDocumentId}, {SimilarDocumentMap.Similarity}) " +
+                        $"values {parameterString} returning {SimilarDocumentMap.MainDocumentId}",
+                    parameters);
+                allInsertedIds = allInsertedIds.Concat(insertedIds);
+            }
+            transaction.Commit();
+        }
+        catch (Exception e)
+        {
+            transaction.Rollback();
+            _logger.LogError(e, "Failed to insert word ratios, rolling back transaction");
+            throw;
+        }
+        return allInsertedIds;
+    }
+
+    public async Task<int> Delete(long mainDocumentId, long similarDocumentId)
```

Comment:
I får nok brug for en delete metode, som kan fjerne mere end bare ét document. Jeg går ud fra at i vil slette alle similar documents, når de skal opdateres, når i ikke har en update function.